src/BatteryStateOfChargeService.cpp

```diff
@@ -16,21 +23,108 @@ BatteryStateOfChargeService::~BatteryStateOfChargeService()
 
 double BatteryStateOfChargeService::totalAmpHoursUsed() const
 {
-    return 0.0;
+    return BATTERY_AMP_HOUR_CAPACITY-ampHours ;
 }
 
 bool BatteryStateOfChargeService::isCharging() const
 {
+    if(newCurrent<0)
+    {
     return false;
+    }
+    else
+    {
+       return true;
+    }
 }
 
 QTime BatteryStateOfChargeService::timeWhenChargedOrDepleted() const
 {
-    return QTime::currentTime();
+    return t;
 }
 
 void BatteryStateOfChargeService::addData(const BatteryData& batteryData)
 {
     Q_UNUSED(batteryData);
     // Update your variables here.
+    initialCurrent = newCurrent;
```

Comment:
using `initial` for these variables (`initialCurrent` and `initialTime`) doesn't seem to be correct as that is not how you are using the variables. `initial` would seem better fit for something on construction of the object, not something you change every time you add data.